### PR TITLE
Retire some old initializer futures

### PR DESCRIPTION
**Pull request by benharsh**
This PR updates and retires some old initializer futures that started working at some point in the past:
- types/records/init/condFieldType.chpl (updated to not use old super.init pattern)
- classes/initializers/records/generics/fullySpecifiedAssignment/noDefaultInitMatch.chpl (don't use super.init, don't delete owned)
- classes/initializers/records/where-clause4b.chpl (don't use super.init)
- classes/initializers/phase1/grandparentAccess.chpl (shift fields after super.init)
- classes/initializers/generics/phase1/omittedCastEtc.chpl (started working at some point, but was using 'delete' on an owned class)

### CODE REVIEW

File: test/classes/initializers/phase1/grandparentAccess.chpl [hunk 1]
```diff
@@ -10,21 +10,20 @@ class B : A {
...
+    b = bVal;
   }
 }
 
 class C : B {
   var c = 4;
 
   proc init(cVal, bVal, aVal) {
-    c = a;
+    var x = a;
```
Comment (lydia-duncan) [inline test/classes/initializers/phase1/grandparentAccess.chpl:22]: Let's leave a comment on this access in the file itself, so some future reader doesn't try to remove it due to thinking it's unnecessary to the purpose of the test

File: test/classes/initializers/records/where-clause4b.future [hunk 2]
```diff
@@ -1,12 +0,0 @@
...
-At the moment, this test outputs an additional number of potential matches that
-were rejected.  All of these appear to be generated for iterators, and are
-likely not "proper initializers".  That is one problem.  This also makes it
-apparent that users which write an incorrect call to an initializer are likely
-to getting increasingly more verbose and unintelligible error messages as we
-convert our library types to use the new initializers.  Something should be done
-to limit the number of candidates we show in such error cases.
-
-When this is fixed, remove this version of the test and the prediff on
-where-clause4.chpl
```
Comment (lydia-duncan) [inline test/classes/initializers/records/where-clause4b.future:12]: This says that the version of the test and the prediff on where-clause4 should be removed when the future is removed

**Review by lydia-duncan — APPROVED** [review]
Just a couple of improvement suggestions that I think will be uncontroversial.  Thanks for noticing these and cleaning them up!

File: test/classes/initializers/records/where-clause4b.future (hunk 2 above)
Comment (benharsh) [reply test/classes/initializers/records/where-clause4b.future:12]: Good catch!

**Pull request merged**